### PR TITLE
move ensure roles

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Move ensure functionality into the resources package to allow usage from other controllers.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/cluster/resources.go [hunk 1]
```diff
@@ -257,35 +257,7 @@ func (cc *Controller) ensureRoles(c *kubermaticv1.Cluster) error {
...
-		}
-
-		if resources.DeepEqual(role, existing) {
-			continue
-		}
-
-		if _, err = cc.kubeClient.RbacV1().Roles(c.Status.NamespaceName).Update(role); err != nil {
-			return fmt.Errorf("failed to update Role %s: %v", role.Name, err)
-		}
+		resources.EnsureRole(data, create, cc.roleLister.Roles(c.Status.NamespaceName), cc.kubeClient.RbacV1().Roles(c.Status.NamespaceName))
```
Comment (kdomanski) [inline api/pkg/controller/cluster/resources.go]:
```
if err = ...; err != nil ; {
  return err
}
```

**Review by kdomanski — CHANGES_REQUESTED** [review]

File: api/pkg/controller/cluster/resources.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/resources.go]: Oh i'm an idiot

Comment (mrIncompetent) [reply api/pkg/controller/cluster/resources.go]: Shouldn't our linting catch that?

Comment (mrIncompetent) [reply api/pkg/controller/cluster/resources.go]: Oh and it did. Embarrassing.

**Review by kdomanski — APPROVED** [review]

**Pull request merged**